address/mock.go

```diff
@@ -14,8 +14,15 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+// RandProofCourierAddr returns a proof courier address with fields populated
+// with valid but random values.
+func RandProofCourierAddr() ProofCourierAddr {
+	return ProofCourierAddr("hashmail://rand.hashmail.proof.courier:443")
```

Comment:
I think we'd want to make the addr string into a richer type so it can handle encoding/decoding these strings itself. See above where I linked to a stdlib function that should prove to be useful.